websocket.go

```diff
@@ -436,6 +423,154 @@ func (c *Client) SendProtobuf(dst *network.ServerIdentity, msg interface{}, ret
 	return nil
 }
 
+// ParallelOptions defines how SendProtobufParallel behaves. Each field has a default
+// value that will be used if 'nil' is passed to SendProtobufParallel. For integers,
+// the default will also be used if the integer = 0.
+type ParallelOptions struct {
+	// Parallel indicates how many requests are sent in parallel.
+	//   Default: half of all nodes in the roster
+	Parallel int
+	// AskNodes indicates how many requests are sent in total.
+	//   Default: all nodes in the roster, except if StartNodes is set > 0
+	AskNodes int
+	// StartNode indicates where to start in the roster. If StartNode is > 0 and < len(roster),
+	// but AskNodes is 0, then AskNodes will be set to len(Roster)-StartNode.
+	//   Default: 0
+	StartNode int
+	// QuitError - if true, the first error received will be returned.
+	//   Default: false
+	QuitError bool
+	// IgnoreNodes is a set of nodes that will not be contacted. They are counted towards
+	// AskNodes and StartNode, but not contacted.
+	//   Default: false
+	IgnoreNodes []*network.ServerIdentity
```

Comment:
hmm.. I wonder if it'll be easier to just use a list of server identities instead of a roster in `SendProtobufParallel`. we give the caller the responsibility of creating a list of nodes that it wants to use in any way he wants so that the caller is not forced to use the `IgnoreNodes` convention in `ParallelOptions`. but the other options (int and bool ones) in `ParallelOptions` are nice to keep.

Reply:
I like the `IgnoreNodes` because it's a simple fix for most of the `GetBlock` calls in byzcoin. I used it in the skipchain-api to exclude the calling node. Example how this looks for the client:
https://github.com/dedis/cothority/pull/1984/files#diff-dde946dfedf27eeee85bb4b483dc2687R998
and the api call:
https://github.com/dedis/cothority/pull/1984/files#diff-55e058d9e0138dcc35f85d083a985a9bR256